### PR TITLE
API stability annotations for ray commands

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Annotate ray commands that are intended to be public. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
I agreed with this selections, but let's see @ericl 's response!

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -1406,6 +1413,7 @@ def rsync_up(cluster_config_file, source, target, cluster_name, all_nodes):
 )
 @click.argument("script_args", nargs=-1)
 @add_click_logging_options
+@PublicAPI
 def submit(
```
Comment (ericl) [inline python/ray/scripts/scripts.py:1417]: Hmm this might be deprecated by jobs. @edoakes @simon-mo should we still recommend submit?

Comment (edoakes) [reply python/ray/scripts/scripts.py:1417]: I don't think we should be recommending it in the post-jobs era, but jobs is not yet GA (until probably 1.13) and I'm sure we have a lot of people using this command

maybe we can leave it not documented as public api given that we intend to deprecate?

**Comment by simon-mo** [comment]
btw I think annotation doesn't do anything to inform the user about the API status? The decorator only changes the docstring, which is not shown to user for CLI command. So the only way user know it is a public API is through reading the source code.

**Review by ericl — APPROVED** [review]

**Comment by rkooo567** [comment]
Failed tests are unrelated

**Pull request merged**